pandas/core/arrays/base.py

```diff
@@ -37,6 +37,41 @@
 
 _extension_array_shared_docs: Dict[str, str] = dict()
 
+_HANDLED_FUNCTIONS = {}
+
+
+def implements(numpy_function) -> Callable[[F], F]:
+    """
+    Register an __array_function__ implementation for ExtensionArray objects.
+    """
+
+    def decorator(func):
+        _HANDLED_FUNCTIONS[numpy_function] = func
+        return func
+
+    return decorator
+
+
+class ArrayFunctionMixin:
+    def __array_function__(self, func, types, args, kwargs):
+        if func not in _HANDLED_FUNCTIONS:
```

Comment:
is _HANDLED_FUNCTIONS supposed to be global or class-specific?

Reply:
never mind, i think i get it

Reply:
I think that needs discussion. it is global here in order to get the basic scaffold in place to implement \_\_array_function__ and get ci to pass (ex. https://github.com/pandas-dev/pandas/pull/35032#issuecomment-691866304) . but for consistency with \_HANDLED_TYPES in \_\_array_ufunc__ we may want to make this class-specific to allow decorating array class methods or have class specific compat methods. concat_compat is compatible with all extensions arrays so decorating this just works (ex. https://github.com/pandas-dev/pandas/pull/35032#issuecomment-691866304).